src/expressions/visitors/EvalVisitor.cpp

```diff
@@ -36,6 +37,17 @@ EvalVisitor::EvalVisitor(EvaluationContext context,
 {
 }
 
+EvalVisitor::EvalVisitor(EvaluationContext context,
+                         Optimisation::LinearProblemApi::FillContext fillContext,
+                         const ModelerStudy::SystemModel::Component* component,
```

Comment:
timeIndex is only needed to evaluate variable. i think it should not be in ctor. this makes the class difficult to understand. Instead you could evaluate the variable for all timesteps (...) in fillcontext